bundler/lib/bundler/cli/gem.rb

```diff
@@ -190,14 +190,23 @@ def run
 
       templates.merge!("exe/newgem.tt" => "exe/#{name}") if config[:exe]
 
-      if extension
+      if extension == "c"
         templates.merge!(
-          "ext/newgem/extconf.rb.tt" => "ext/#{name}/extconf.rb",
+          "ext/newgem/extconf-c.rb.tt" => "ext/#{name}/extconf.rb",
           "ext/newgem/newgem.h.tt" => "ext/#{name}/#{underscored_name}.h",
           "ext/newgem/newgem.c.tt" => "ext/#{name}/#{underscored_name}.c"
         )
       end
 
+      if extension == "rust"
+        templates.merge!(
+          "Cargo.toml.tt" => "Cargo.toml",
+          "ext/newgem/Cargo.toml.tt" => "ext/#{name}/Cargo.toml",
+          "ext/newgem/extconf-rust.rb.tt" => "ext/#{name}/extconf.rb",
+          "ext/newgem/src/lib.rs.tt" => "ext/#{name}/src/lib.rs",
```

Comment:
Should we use `newgem.rs.tt` same as C extension?

Reply:
I was thinking about renaming folder `ext/newgem` to `ext/newgem-c` and `ext/newgem-rust` or similar, but not all files related to extensions are in `ext` folder :thinking:.

Reply:
Ahh, you mean for the file itself. `src/lib.rs` is standard Cargo name of Rust library entry point.

https://doc.rust-lang.org/book/ch07-01-packages-and-crates.html

> After we run cargo new, we use ls to see what Cargo creates. In the project directory, there’s a Cargo.toml file, giving us a package. There’s also a src directory that contains main.rs. Open Cargo.toml in your text editor, and note there’s no mention of src/main.rs. Cargo follows a convention that src/main.rs is the crate root of a binary crate with the same name as the package. Likewise, Cargo knows that if the package directory contains src/lib.rs, the package contains a library crate with the same name as the package, and src/lib.rs is its crate root. Cargo passes the crate root files to rustc to build the library or binary.

> Here, we have a package that only contains src/main.rs, meaning it only contains a binary crate named my-project. If a package contains src/main.rs and src/lib.rs, it has two crates: a binary and a library, both with the same name as the package. A package can have multiple binary crates by placing files in the src/bin directory: each file will be a separate binary crate.

Reply:
Thanks, I see. `lib.rs` is reasonable.